core/src/plan/primary_key.rs

```diff
@@ -389,6 +389,142 @@ mod tests {
         assert_eq!(actual, expected, "AND binary op 3:\n{sql}");
     }
 
+    #[test]
+    fn where_expr_multiple_primary_keys() {
+        let storage = run("
+            CREATE TABLE Player (
+                id INTEGER,
+                name TEXT,
+                PRIMARY KEY (id, name)
+            );
+        ");
+
+        let sql = "SELECT * FROM Player WHERE id = 1;";
+        let actual = plan(&storage, sql);
+        let expected = select(Select {
+            projection: vec![SelectItem::Wildcard],
+            from: TableWithJoins {
+                relation: TableFactor::Table {
+                    name: "Player".to_owned(),
+                    alias: None,
+                    index: None,
```

Comment:
if the primary key planner works, then `index` field should exist.
based on that `index` field, execution layer handles whether to call `Store::fetch_data` or `Store::scan_data`.

Reply:
Ok, I will add this to the documentation of the planner.

Reply:
Should the primary key planner work for select statements that are not operating on a `TableFactor::Table`? Without knowing which table is involved, we cannot know which expression entries are the primary keys as we cannot extract their schema.

Reply:
> Should the primary key planner work for select statements that are not operating on a TableFactor::Table

we only need to handle `TableFactor::Table`. we can ignore other TableFactor variants.

Reply:
Can you point me to where the index is used afterwards? I am still trying to understand how to represent the multi-primary key variant and seeing how it is used may certainly help.